compiler/cpp/src/thrift/generate/t_generator.h

```diff
@@ -96,7 +97,33 @@ class t_generator {
     return escape_string(constval->get_string());
   }
 
+  /**
+   * Check if all identifiers are valid for the target language
+   */
+  virtual void validate_input() const;
+
 protected:
+  virtual std::set<std::string> lang_keywords() const;
+
+  /**
+   * A list of reserved words that cannot be used as identifiers.
+   */
+  const std::set<std::string> keywords_;
+
+  virtual void validate_id(const std::string& id) const;
+
+  virtual void validate(t_enum const* en) const;
```

Comment:
validate looks like an opportunity to use template specialization, since half of them do the same thing.  Just a thought. :)